package errors

import (
	"log"

	"{{.App.Repository}}/pkg/validator"
	"github.com/gofiber/fiber/v2"
)

const (
	DefaultError = "Internal server error"
)

func Handler(ctx *fiber.Ctx, err error) error {
	if e, ok := err.(*validator.ValidationError); ok {
		return ctx.Status(e.Code).JSON(e)
	}

	if e, ok := err.(*fiber.Error); ok {
		return ctx.Status(e.Code).JSON(e)
	}

	log.Default().Println(err)

	return ctx.Status(fiber.StatusInternalServerError).JSON(fiber.ErrInternalServerError)
}
